src/swarm/neo/client/mixins/BatchRequestCore.d

```diff
@@ -0,0 +1,257 @@
+      the standard logic for a request controller accessible via the user API.
+      Has public methods suspend(), resume(), and stop().
+
+    * BatchRequestSharedWorkingData: struct encapsulating shared working
+      data required by the helpers in this module. If using these helpers, a
+      field of type BatchRequestSharedWorkingData, named
+      suspendable_control must be added to the request's shared working data
+      struct.
+
+    Copyright: Copyright (c) 2017 sociomantic labs GmbH. All rights reserved
+
+    License:
+        Boost Software License Version 1.0. See LICENSE_BOOST.txt for details.
+
+*******************************************************************************/
+
+module swarm.neo.client.mixins.BatchRequestCore;
+
+import swarm.neo.client.RequestOnConn;
+import swarm.neo.client.mixins.AllNodesRequestCore;
+
+/*******************************************************************************
+
+    Mixes in a controller class for a request (assumed to contain the features
+    of swarm.neo.client.mixins.RequestCore : RequestCore) which is suspendable,
+    resumable, and stoppable. The class implements the interface IController
+    (specified as a template argument), assumed to have methods suspend(),
+    resume(), and stop(). (Note that this interface is not provided in the
+    library purely in order to keep all API definitions for each request inside
+    a single module, not scattered around different modules in swarm.)
+
+    Params:
+        Request = type of request struct
+        IController = interface defining the controller API as presented to the
+            user. Assumed to contain methods suspend(), resume(), and stop() (in
+            addition to any other methods required by the request).
+        MessageType = type of enum defining the types of messages used by this
+            request's protocol. Expected to contain the members Suspend, Resume,
+            and Stop (in addition to any others required by the request).
+
+*******************************************************************************/
+
+public template BatchController ( Request, IController )
+{
+    /***************************************************************************
+
+        Request controller, accessible to the user via the client's `control()`
+        method.
+
+    ***************************************************************************/
+
+    public scope class Controller : IController
+    {
+        import ocean.core.Enforce;
+        import swarm.neo.client.mixins.RequestCore : ControllerBase;
+
+        /***********************************************************************
+
+            Base mixin.
+
+        ***********************************************************************/
+
+        mixin ControllerBase;
+
+        /***********************************************************************
+
+            Tells the nodes to stop sending data to this request.
```

Comment:
These "Tells the nodes" comments are misleading now. Maybe change to something like "Stops the flow of data to this request."?